google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigtableColumn.java

```diff
@@ -0,0 +1,203 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery;
+
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.Objects;
+
+public class BigtableColumn implements Serializable {
+
+
+  static final Function<com.google.api.services.bigquery.model.BigtableColumn, BigtableColumn> FROM_PB_FUNCTION =
+      new Function<com.google.api.services.bigquery.model.BigtableColumn, BigtableColumn>() {
+        @Override
+        public BigtableColumn apply(
+            com.google.api.services.bigquery.model.BigtableColumn pb) {
+          return BigtableColumn.fromPb(pb);
+        }
+      };
+
+  static final Function<BigtableColumn, com.google.api.services.bigquery.model.BigtableColumn> TO_PB_FUNCTION =
+      new Function<BigtableColumn, com.google.api.services.bigquery.model.BigtableColumn>() {
+        @Override
+        public com.google.api.services.bigquery.model.BigtableColumn apply(
+            BigtableColumn column) {
+          return column.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 1L;
+
+  public String getQualifierEncoded() { return qualifierEncoded; }
+  public String getFieldName() { return fieldName; }
+  public Boolean getOnlyReadLatest() { return onlyReadLatest; }
+  public String getEncoding() { return encoding; }
+  public String getType() { return type; }
+
+  private final String qualifierEncoded;
+  private final String fieldName;
+  private final Boolean onlyReadLatest;
+  private final String encoding;
+  private final String type;
+
+  static final class Builder {
+    private String qualifierEncoded;
+    private String fieldName;
+    private Boolean onlyReadLatest;
+    private String encoding;
+    private String type;
+
+    private Builder() {}
+
+    /**
+     * Qualifier of the column.
+     *
+     * Columns in the parent column family that has this exact qualifier are exposed
+     * as . field. If the qualifier is valid UTF-8 string, it can be specified in the
+     * qualifier_string field. Otherwise, a base-64 encoded value must be set to qualifier_encoded. The column field name is the same as the
+     * column qualifier. However, if the qualifier is not a valid BigQuery field
+     * identifier, a valid identifier must be provided as field_name.
+     */
+    Builder setQualifierEncoded(String qualifierEncoded) {
+      this.qualifierEncoded = qualifierEncoded;
```

Comment:
Could you have used AutoValue for this class? It's looking pretty boilerplatey to me.

Reply:
Swapped.  I should just embrace autovalue by default, instead of the "i may need something clever down the line".